enhancements/profile-versioning.md

```diff
@@ -0,0 +1,245 @@
+
+The Compliance Operator currently supports upgrades, where fixes are rarely, if
+ever, backported. Due to this, the Compliance Operator does not support
+downgrades. The operator will upgrade automatically if configured to do so. To
+best support users, the upgrade to a version of the Compliance Operator with
+this enhancement must do one of two things.
+
+1. Version all supported profiles and make `version` a required property of the `Profile` CRD
+2. Make `version` an optional property, and version profiles incrementally
+
+In either case, it is expected that user can gracefully upgrade to a release of
+the Compliance Operator that includes this enhancement.
+
+Once a profile is versioned, it is expected to remain versioned for its
+life cycle. This expectation matches reality where profiles are version in some
+way from an authoring body (e.g., PCI-DSS 3.2.1 to 4.0.0, or CIS OpenShift
+1.3.0 to 1.4.0).
+
+### Version Skew Strategy
+
+Not applicable to this particular enhancement given the Compliance Operator is
+a single installation within a single cluster.
+
+### Operational Aspects of API Extensions
+
+Operational impact on this change should be minimal considering the profiles
+need to be parsed regardless, and this change is limited to setting one
+additional property on a CRD.
+
+The compliance content shipped with the Compliance Operator by default may
+change, which will still be true with this enhancement. Users should expect to
+use `Profile` CRDs in the same way they have in the past.
+
+#### Failure Modes
+
+A potential failure mode would be if `version` were a required property of the
+`Profile` CRD and the data stream contained a profile that didn't have a
+version. In that case, profile parsing would fail, and not allow the user to
+use the operator at all. This particular failure could be prevented in CI by
+ensuring each profile contains a version, if it is a required property.
+
+#### Support Procedures
+
+In the case above, users supplying their own content must ensure they're
+setting the profile version in the data stream. Otherwise, they will not be
+able to use custom content with the compliance operator.
+
+Verbose logging will be added to the profile parser to detail the `Profile`
+CRDs being created, and the versions parsed from the data stream so these
+failure scenarios are traceable, and easy to detect.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+1. Update each profile version for OpenShift profiles in [ComplianceAsCode/content](https://github.com/ComplianceAsCode/content)
+2. Implement a new property called `version` for the `Profile` CRD
+3. Add a CI end-to-end test that fails if a profile does not contain a `version`
+4. Document profile life cycles across content and the operator, including versioning
+
+## Drawbacks
+
+A potential drawback is how to best maintain versions across profiles, or if
```

Comment:
I think that depends on how we decide to implement versioning, and it could also depend on the benchmark.

Users are typically pushing us to update CIS benchmarks before we get around to it, so they're usually ok with the automatic update.

PCI-DSS is a different story though, where users need plan of action and milestones for getting from 3.2.1 compliance to 4.0.0 compliance. In these cases, I'm inclined to offer a PCI-DSS profile for each version, which would mean if they scan for PCI-DSS 3.2.1, it will continue to work even after we ship PCI-DSS 4.0.0.

The customer can run both profiles as separate scans, and ideally converge the results as they move towards a PCI-DSS 4.0.0 compliant environment.

Does that sound reasonable?